File: src/main/java/com/dh/ondot/schedule/domain/enums/SoundCategory.java
package com.dh.ondot.schedule.domain.enums;

import com.dh.ondot.core.exception.UnsupportedException;

import java.util.Locale;

import static com.dh.ondot.core.exception.ErrorCode.UNSUPPORTED_SOUND_CATEGORY;

public enum SoundCategory {
    ;

    public static SoundCategory from(String category) {
        try {
            return SoundCategory.valueOf(category.toUpperCase(Locale.ENGLISH));
        } catch (IllegalArgumentException e) {
            throw new UnsupportedException(UNSUPPORTED_SOUND_CATEGORY, category);
        }
    }
}

File: src/main/kotlin/com/dh/ondot/schedule/domain/enums/SoundCategory.kt
package com.dh.ondot.schedule.domain.enums

import com.dh.ondot.core.exception.ErrorCode.UNSUPPORTED_SOUND_CATEGORY
import com.dh.ondot.core.exception.UnsupportedException
import java.util.Locale

enum class SoundCategory {
    BRIGHT_ENERGY,
    FAST_INTENSE;

    companion object {
        @JvmStatic
        fun from(category: String): SoundCategory =
            try {
                valueOf(category.uppercase(Locale.ENGLISH))
            } catch (e: IllegalArgumentException) {
                throw UnsupportedException(UNSUPPORTED_SOUND_CATEGORY, category)
            }
    }
}
